components/tel/Core/Src/radio.c

```diff
@@ -65,13 +73,59 @@ static void copy_CAN_id(uint8_t* source, CAN_Radio_msg_t *tx_CAN_msg, uint32_t c
 
 
 /**
- * @brief Sends a CAN message over the radio by packing it into a CAN_BUFFER_LEN buffer 
- *        and transmits it over UART
+ * @brief Takes api_buffer and packages it into an api frame. transmits this api frame over uart.
+ * @param radio_buffer[]: pointer to an array in which the messages are being added, api_buffer_postion: position in api_buffer where messages end.
+ * @return void
+*/
+void RADIO_tx_API_Packager(uint8_t api_buffer[], uint16_t api_buffer_position)
+{
+ uint16_t api_packet_size = API_OVERHEAD_SIZE + api_buffer_position;
+ uint8_t api_packet[api_packet_size];
+ XBEE_api_overhead_setup(api_packet);
+
+ for (int i = 0; i<api_buffer_position; i++){
+	 api_packet[MESSAGE_DATA_START_POSITION + i] = api_buffer[i];
+ }
+
+ XBEE_calculate_length(api_packet, api_packet_size);
+ XBEE_calculate_checksum(api_packet, api_packet_size);
+
+ HAL_UART_Transmit(&huart1, api_packet, sizeof(api_packet), CAN_TRANSMIT_TIMEOUT);                     /* Transmit over UART */
+}
+
+
+
+/**
+ * @brief Takes CAN radio buffer and adds it to an API buffer Array. Once enough messages are in the array, array is sent for packaging.
+ * @param radio_buffer[]: pointer to an array in which the messages are being added.
+ * @return void
+*/
+void RADIO_tx_API_Accumulator(uint8_t radio_message[])
+{
+  int i;
+  for (i = 0; i<CAN_BUFFER_LEN; i++){
+	  api_buffer[i+api_buffer_position] = radio_message[i];
+  }
+
+  api_message_count++;
+  api_buffer_position = api_buffer_position + i;
+
+  if (api_message_count == API_PACKET_SIZE){
```

Comment:
Consider changing API_PACKET_SIZE to API_MAX_MSGS or something to indicate that this is the number of CAN messages in a Packet.